openhands/controller/agent_controller.py

```diff
@@ -121,8 +129,34 @@ def __init__(
         self._stuck_detector = StuckDetector(self.state)
 
     async def close(self):
-        """Closes the agent controller, canceling any ongoing tasks and unsubscribing from the event stream."""
+        """Closes the agent controller, canceling any ongoing tasks and unsubscribing from the event stream.
+
+        Note that it's fairly important that this closes properly, otherwise the state is incomplete."""
         await self.set_agent_state_to(AgentState.STOPPED)
+
+        # we made history, now is the time to rewrite it!
```

Comment:
I don't _love_ this 🙃 would it be possible (maybe in the future rather than in this PR) to have these external consumers use the eventstream instead of `history`?